receiver/prometheusreceiver/internal/metricfamily_test.go

```diff
@@ -293,6 +294,199 @@ func TestMetricGroupData_toDistributionUnitTest(t *testing.T) {
+				attributes := point.Attributes()
+				attributes.PutStr("a", "A")
+				attributes.PutStr("b", "B")
+				return point
+			},
+		},
+		{
+			name:                "integer histogram with startTimestamp from _created",
+			metricName:          "request_duration_seconds",
+			intervalStartTimeMs: 11,
+			labels:              labels.FromMap(map[string]string{"a": "A"}),
+			scrapes: []*scrape{
+				{
+					at:     11,
+					metric: "request_duration_seconds",
+					integerHistogram: &histogram.Histogram{
+						CounterResetHint: histogram.UnknownCounterReset,
+						Schema:           1,
+						ZeroThreshold:    0.42,
+						ZeroCount:        1,
+						Count:            66,
+						Sum:              1004.78,
+						PositiveSpans:    []histogram.Span{{Offset: 1, Length: 2}, {Offset: 3, Length: 1}},
+						PositiveBuckets:  []int64{33, -30, 26}, // Delta encoded counts: 33, 3=(33-30), 30=(3+27) -> 65
+						NegativeSpans:    []histogram.Span{{Offset: 0, Length: 1}},
+						NegativeBuckets:  []int64{1}, // Delta encoded counts: 1
+					},
+				},
+				{
+					at:     11,
+					metric: "request_duration_seconds_created",
+					value:  600.78,
+				},
+			},
+			want: func() pmetric.ExponentialHistogramDataPoint {
+				point := pmetric.NewExponentialHistogramDataPoint()
+				point.SetCount(66)
+				point.SetSum(1004.78)
+				point.SetTimestamp(pcommon.Timestamp(11 * time.Millisecond)) // the time in milliseconds -> nanoseconds.
+				point.SetStartTimestamp(timestampFromFloat64(600.78))        // the time in milliseconds -> nanoseconds.
+				point.SetScale(1)
+				point.SetZeroThreshold(0.42)
+				point.SetZeroCount(1)
+				point.Positive().SetOffset(0)
+				point.Positive().BucketCounts().FromRaw([]uint64{33, 3, 0, 0, 0, 29})
+				point.Negative().SetOffset(-1)
+				point.Negative().BucketCounts().FromRaw([]uint64{1})
+				attributes := point.Attributes()
+				attributes.PutStr("a", "A")
+				return point
+			},
+		},
+		{
+			name:                "integer histogram that is stale",
+			metricName:          "request_duration_seconds",
+			intervalStartTimeMs: 11,
+			labels:              labels.FromMap(map[string]string{"a": "A", "b": "B"}),
+			scrapes: []*scrape{
+				{
+					at:     11,
+					metric: "request_duration_seconds",
+					integerHistogram: &histogram.Histogram{
+						Sum: math.Float64frombits(value.StaleNaN),
+					},
+				},
+			},
+			want: func() pmetric.ExponentialHistogramDataPoint {
+				point := pmetric.NewExponentialHistogramDataPoint()
+				point.SetTimestamp(pcommon.Timestamp(11 * time.Millisecond)) // the time in milliseconds -> nanoseconds.
+				point.SetFlags(pmetric.DefaultDataPointFlags.WithNoRecordedValue(true))
+				point.SetStartTimestamp(pcommon.Timestamp(11 * time.Millisecond)) // the time in milliseconds -> nanoseconds.
+				attributes := point.Attributes()
+				attributes.PutStr("a", "A")
+				attributes.PutStr("b", "B")
+				return point
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
```

Comment:
Is this line redundant?

Reply:
Isn't the loop var fix in Go 1.22 ? We require 1.21 right now. Maybe I'm mixing this up, but there's 3 other tests that have this line (I did copy :) ).